gprofiler/utils.py

```diff
@@ -85,19 +85,26 @@ def start_process(cmd: Union[str, List[str]], **kwargs) -> Popen:
     return popen
 
 
-def poll_process(process, timeout, stop_event):
-    timefn = time.monotonic
-    endtime = timefn() + timeout
+def wait_event(timeout: float, stop_event: Event, condition: Callable[[], bool]) -> bool:
```

Comment:
I think raising a `TimeoutError` on timeout is clearer than returning a boolean.

Reply:
Aight